### ISSUE TITLE
Exception when reading attribute data

### ISSUE THREAD

**Issue by Octogonapus** [jamesmudd/jhdf#101]
I have this code to read an attribute of an HDF5:
```kotlin
HdfFile(file).use {
    val config = it.getAttribute("model_config").data as String
    // use config...
}
```

jhdf is able to read the attribute (i.e. `it.getAttribute("model_config")`) but reading its data (i.e. `.data`) throws this exception: https://pastebin.com/i6U9mc1R. Maybe it's worth noting that other tools (e.x., Netron) can load the file without errors, so I don't think there is a problem with the file.

I have attached the HDF5 file to this issue. I have also attached the log file. I am using jhdf version 0.4.6.
[model2.zip](https://github.com/jamesmudd/jhdf/files/3613898/model2.zip)
[axon_20190915110753.log](https://github.com/jamesmudd/jhdf/files/3613902/axon_20190915110753.log)

**Comment by Octogonapus** [comment]
This is the program state right before the exception is thrown (throw on `headerData.limit(length)`:
![image](https://user-images.githubusercontent.com/4064722/64923696-64f99080-d7aa-11e9-85f2-0bf7ce51de8d.png)

Looks like what happens is the program tries to take a slice of a `source`, which is already full, so the size of the slice is zero.

**Comment by Octogonapus** [comment]
Before this line (GlobalHeap:92) `int size = readBytesAsUnsignedInt(bb, hdfFc.getSizeOfOffsets());` the remaining size of `bb` is `8`. Reading the size takes the last `8`, making the remaining size of `bb` become `0`.

This method is guarded by this condition on the loop when the heap is read into memory (GlobalHeap:63) `while (bb.remaining() > 8)`, maybe this check should be at least `12` instead of `8`?

**Comment by jamesmudd** [comment]
Thanks very much for the detailed report. I have reproduced the issue using your file and this does look like a `jhdf` bug. I need to spend some more time investigating but will hopefully have a fix soon.

**Comment by jamesmudd** [comment]
Have just released v0.4.7 which includes the fix for this issue. Hopefully this will now work fine for you let me know. Thanks

**Comment by Octogonapus** [comment]
Bug fixed and no regressions I can find 👍 
Thanks for the quick fix!
